sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler_opencl.asciidoc

```diff
@@ -107,6 +112,127 @@ enum class source_language : /*unspecified*/ {
 } // namespace sycl::ext::oneapi::experimental
 ```
 
+=== Queries
+
+==== Version type
+
+This extension adds the following type and constant definitions, which help
+identify the version of OpenCL and its extensions.
+
+|====
+a|
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+namespace sycl::ext::oneapi::experimental {
+
+struct cl_version {
+  unsigned major:10;
+  unsigned minor:10;
+  unsigned patch:12;
+};
+
+inline constexpr cl_version opencl_c_1_0 = {1,0,0};
+inline constexpr cl_version opencl_c_1_1 = {1,1,0};
+inline constexpr cl_version opencl_c_1_2 = {1,2,0};
+inline constexpr cl_version opencl_c_2_0 = {2,0,0};
+inline constexpr cl_version opencl_c_3_0 = {3,0,0};
```

Comment:
There is an example at the bottom of the spec under "Querying supported features and extensions".  Search for "opencl_c_3_0".

I think the shorthands are better than the template because they provide some guidance on the set of valid OpenCL C versions.  This is particularly tricky because the OpenCL C versions are not exactly the same as the OpenCL versions.  For example, there is no such thing as OpenCL C 2.1 or 2.2.  If we just provided the template, people might write code like this:

```
if (d.ext_oneapi_supports_cl_c_version(syclex::opencl_c<2,2>) {/*...*/}
```

This would compile without error, but the comparison would always be false because there is no such thing as OpenCL C 2.2.

I added a non-normative note about the OpenCL C version being different from the OpenCL version in 15193cff1dca915f37f02b59ee9bbe7a5f272e57.

Reply:
> There is an example at the bottom of the spec under "Querying supported features and extensions". Search for "opencl_c_3_0".

Thanks, and sorry for missing that.

> I think the shorthands are better than the template because they provide some guidance on the set of valid OpenCL C versions. This is particularly tricky because the OpenCL C versions are not exactly the same as the OpenCL versions

Ah, okay. I didn't appreciate the difference here.  The non-normative note really helps, and I agree the shorthands are better if the intent really is to define an exhaustive list.